core/XH.js

```diff
@@ -147,6 +147,9 @@ class XHClass {
     /** State of app - see AppState for valid values. */
     @observable appState = AppState.PRE_AUTH;
 
+    /** Milliseconds since last detected user activity */
+    lastActivityMs = Date.now();
```

Comment:
should probably be a getter backed by an _lastActivityMs. in order to make it read only